src/code/z_skin_matrix.c

```diff
@@ -1,42 +1,692 @@
+    cos = Math_Coss(roll);
+    mf->xx = cos;
+    mf->xz = -sin;
+    mf->zw = 0;
+    mf->yw = 0;
+    mf->xw = 0;
+    mf->wz = 0;
+    mf->wy = 0;
+    mf->wx = 0;
+    mf->ww = 1;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_skin_matrix/func_800A7A24.s")
+    if (yaw != 0) {
+        sin2 = Math_Sins(yaw);
+        cos2 = Math_Coss(yaw);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_skin_matrix/func_800A7A84.s")
+        mf->zz = cos * cos2;
+        mf->yz = cos * sin2;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_skin_matrix/func_800A7B04.s")
+        mf->zx = sin * cos2;
+        mf->yx = sin * sin2;
+        mf->zy = -sin2;
+        mf->yy = cos2;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_skin_matrix/func_800A7B84.s")
+    } else {
+        mf->zz = cos;
+        if (1) {} // required to match
+        if (1) {} // required to match
+        yx = sin; // required to match
+        mf->zx = sin;
+        mf->yx = mf->yz = mf->zy = 0;
+        mf->yy = 1;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_skin_matrix/func_800A7BE4.s")
+    if (pitch != 0) {
+        sin2 = Math_Sins(pitch);
+        cos2 = Math_Coss(pitch);
+        xx = mf->xx;
+        yx = mf->yx;
+        mf->xx = (xx * cos2) + (yx * sin2);
+        mf->yx = yx * cos2 - (xx * sin2);
+        if (1) {} // required to match
+        yz = mf->yz;
+        xz = mf->xz;
+        mf->xz = (xz * cos2) + (yz * sin2);
+        mf->yz = (yz * cos2) - (xz * sin2);
+        if (cos2) {} // required to match
+        mf->xy = mf->yy * sin2;
+        mf->yy = mf->yy * cos2;
+    } else {
+        mf->xy = 0;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_skin_matrix/func_800A7C20.s")
+// Output matrix translates vector by dx in x direction, dy in y direction, dz in z direction
+void SkinMatrix_Translate(MtxF* mf, f32 dx, f32 dy, f32 dz) {
+    mf->xy = 0.0f;
+    mf->xz = 0.0f;
+    mf->xw = 0.0f;
+    mf->yx = 0.0f;
+    mf->yz = 0.0f;
+    mf->yw = 0.0f;
+    mf->zx = 0.0f;
+    mf->zy = 0.0f;
+    mf->zw = 0.0f;
+    mf->xx = 1.0f;
+    mf->yy = 1.0f;
+    mf->zz = 1.0f;
+    mf->ww = 1.0f;
+    mf->wx = dx;
+    mf->wy = dy;
+    mf->wz = dz;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_skin_matrix/func_800A7C60.s")
+// output matrix scales, then rotates (RPY), then translates the vector
+void SkinMatrix_ScaleRotateRPYTranslate(MtxF* mf, f32 scaleX, f32 scaleY, f32 scaleZ, s16 roll, s16 pitch, s16 yaw,
+                                        f32 dx, f32 dy, f32 dz) {
+    MtxF mft1;
+    MtxF mft2;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_skin_matrix/func_800A7E70.s")
+    SkinMatrix_Translate(mf, dx, dy, dz);            // translation matrix
+    SkinMatrix_RotateRPY(&mft1, roll, pitch, yaw);   // rotation matrix
+    SkinMatrix_MtxFMtxFMult(mf, &mft1, &mft2);       // MtxFMtxFMult
+    SkinMatrix_Scale(&mft1, scaleX, scaleY, scaleZ); // scale matrix
+    SkinMatrix_MtxFMtxFMult(&mft2, &mft1, mf);       // MtxFMtxFMult
```

Comment:
these comments seem redundant cause the name of the function being called tells you this info